chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/SurveillanceReportController.java

```diff
@@ -0,0 +1,220 @@
+    public @ResponseBody QuarterlyReport getQuarterlyReport(@PathVariable final Long quarterlyReportId)
+            throws AccessDeniedException, EntityRetrievalException {
+        if (!ff4j.check(FeatureList.SURVEILLANCE_REPORTING)) {
+            throw new NotImplementedException();
+        }
+        QuarterlyReportDTO reportDto = reportManager.getQuarterlyReport(quarterlyReportId);
+        return new QuarterlyReport(reportDto);
+    }
+
+    @ApiOperation(value = "Create a new quarterly surveillance report.",
+                    notes = "An effort will be made to determine which ACB the report belongs to "
+                            + "if one is not provided in the request."
+                            + "Security Restrictions: ROLE_ADMIN or ROLE_ACB and administrative "
+                            + "authority on the ACB associated with the report.")
+    @RequestMapping(value = "/quarterly", method = RequestMethod.POST, produces = "application/json; charset=utf-8")
+    public synchronized QuarterlyReport createQuarterlyReport(
+            @RequestBody(required = true) final QuarterlyReport createRequest) 
+    throws AccessDeniedException, InvalidArgumentsException, EntityCreationException {
+        if (!ff4j.check(FeatureList.SURVEILLANCE_REPORTING)) {
+            throw new NotImplementedException();
+        }
+        if (createRequest.getAcb() == null || (createRequest.getAcb().getId() == null
+                && StringUtils.isEmpty(createRequest.getAcb().getName()))) {
+            //no acb information was provided - check user permissions to see if
+            //we can determine which ACB the user has access to (this is fine if they have only one)
+            List<CertificationBodyDTO> allowedAcbs = resourcePermissions.getAllAcbsForCurrentUser();
+            if (allowedAcbs != null && allowedAcbs.size() == 1) {
+                CertificationBody acb = new CertificationBody(allowedAcbs.get(0));
+                createRequest.setAcb(acb);
+            }
+        } else if (createRequest.getAcb() != null && createRequest.getAcb().getId() == null
+                && !StringUtils.isEmpty(createRequest.getAcb().getName())) {
+            //just an ACB name was provided - fill in the ACB object if possible
+            List<CertificationBodyDTO> allowedAcbs = resourcePermissions.getAllAcbsForCurrentUser();
+            for (CertificationBodyDTO allowedAcb : allowedAcbs) {
+                if (createRequest.getAcb().getName().equals(allowedAcb.getName())) {
+                    CertificationBody acb = new CertificationBody(allowedAcb);
+                    createRequest.setAcb(acb);
+                }
+            }
+        }
+
+        if (createRequest.getAcb() == null || createRequest.getAcb().getId() == null) {
+            throw new InvalidArgumentsException(msgUtil.getMessage("report.quarterlySurveillance.missingAcb"));
+        }
+
+        //create the report
```

Comment:
I've always thought of controllers as being "dumb".  Do you think some of this logic and object building belongs in the Manager?

Reply:
I agree with you and am not sure how to proceed here. This bit of code is part of converting the request object into the DTO object that gets sent into the manager. If an ACB is provided it doesn't do anything, but if one isn't it's checking to see what ACBs the user has access to and if it's only one then it creates the report under that one. If the user has multiple ACBs then it throws an error. I was thinking of what I'm doing here as validation on the request object that's sent in. Do you think the controller is an appropriate place for a check like that or it really should all go into the manager? The manager just works with whatever ACB is passed in with the DTO object.

Reply:
@tmy1313 I've updated this as we discussed.